modules/atomic-widgets/props-resolver/props-resolver.php

```diff
@@ -97,10 +89,32 @@ private function transform( $value, int $depth = 0 ) {
 			return null;
 		}
 
-		// Transform nested transformable values recursively.
-		if ( is_array( $value['value'] ) ) {
+		if ( $prop_type instanceof Union_Prop_Type ) {
+			$prop_type = $prop_type->get_prop_type( $value['$$type'] );
+
+			if ( ! $prop_type ) {
+				return null;
+			}
+		}
+
+		if ( $prop_type instanceof Object_Prop_Type ) {
+			if ( ! is_array( $value['value'] ) ) {
```

Comment:
At that point, isn't the value always an array?

Reply:
This is for an edge case, when the prop type is an array but the value can be any thing that was successfully saved in the DB, really this is just for protection against failing in the Front End